hoodie-common/src/main/java/com/uber/hoodie/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -232,24 +255,43 @@ public void deleteInflight(HoodieInstant instant) {
     return readDataFromPath(detailPath);
   }
 
-  protected void moveInflightToComplete(HoodieInstant inflight, HoodieInstant completed,
+  public void revertFromInflightToRequested(HoodieInstant inflightInstant, HoodieInstant requestedInstant,
+      Optional<byte[]> data) {
+    Preconditions.checkArgument(inflightInstant.getAction().equals(HoodieTimeline.COMPACTION_ACTION));
+    transitionState(inflightInstant, requestedInstant, data);
+  }
+
+  public void transitionFromRequestedToInflight(HoodieInstant requestedInstant, HoodieInstant inflightInstant,
```

Comment:
are we making sure a rollback does not take this inflight into consideration ? is the action driving that ?

Reply:
Rollback/Cleaner/Archiving changes are expected in subsequent PRs.

Reply:
Compaction Inflight is different from inflight commits. Will ensure rollback does not step in inflight compactions